loudness_pcm.c

```diff
@@ -300,8 +302,14 @@ static snd_pcm_sframes_t transfer_callback(
     unsigned int i;
     int j;
 
-    const unsigned int M = context->lengths[0];
-    const unsigned int N = context->lengths[1];
+    if(size > context->period_size) {
+		/* Do not process more than max period size to avoid buffer overruns */
+		/* ALSA can deal with a partially fullfilled transfer just fine */
+		size = context->period_size;
+    }
+
```

Comment:
As I mentioned in my previous comment, this whole check is done already, [here](https://github.com/dpapavas/alsaloudness/blob/master/loudness_pcm.c#L392), so you can remove the `if` block altogether, along with the `period_size` field from `context`, since it won't be necessary any longer.  (It will also keep breaking the plugin if `fft` is set explicitly, as `period_size` will then be left zero.)

Also, the initial commit introducing this (6d8dd2c75cd514e935ec4960bf2ff7fb817a368f), had the log message:

>Fix segfault when filling many periods at once

What was the nature of the segfault it attempted to fix?  Should we be worried about it?

Reply:
>Should we be worried about it?

No, the check is already done somewhere else in the code. I just missed it.